tests/test_pipeline/test_pipeline_all.py

```diff
@@ -0,0 +1,842 @@
+"""
+Based on imbalanced learn pipeline's tests:
+https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/tests/test_pipeline.py
+
+And sklearn's pipeline's tests:
+https://github.com/scikit-learn-contrib/imbalanced-learn/blob/master/imblearn/tests/test_pipeline.py
+"""
+import itertools
+import re
+import shutil
+from tempfile import mkdtemp
+
+import numpy as np
+import pytest
+from joblib import Memory
+from pytest import raises
+from sklearn.base import clone
+from sklearn.cluster import KMeans
+from sklearn.datasets import load_iris, make_classification
+from sklearn.decomposition import PCA
+from sklearn.feature_selection import SelectKBest, f_classif
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.pipeline import FeatureUnion
+from sklearn.preprocessing import StandardScaler
+from sklearn.svm import SVC
+from sklearn.utils._testing import (
+    assert_array_almost_equal,
+    assert_array_equal,
+)
+from sklearn.tests.test_pipeline import (
+NoFit,NoTrans,NoInvTransf,Transf,TransfFitParams,Mult,FitParamT,DummyTransf,DummyEstimatorParams,
+)
+from sklearn.tests.test_pipeline import (
+test_pipeline_invalid_parameters,
+test_pipeline_init_tuple,
+test_pipeline_methods_anova,
+
+)
+from feature_engine.pipeline import Pipeline, make_pipeline
+
+JUNK_FOOD_DOCS = (
+    "the pizza pizza beer copyright",
+    "the pizza burger beer copyright",
+    "the the pizza beer beer copyright",
+    "the burger beer beer copyright",
+    "the coke burger coke copyright",
+    "the coke burger burger",
+)
+
+R_TOL = 1e-4
+
+def test_pipeline_invalid_parameters():
+    test_pipeline_invalid_parameters
+
```

Comment:
@solegalli copying tests from another library like that is generally not best practice because:

- The test functions are not typically designed to be part of a library's public API. This means they might not be as stable or intended for external use
- If scikit-learn updates their tests or implementation in a way that is incompatible with your extension, your tests might suddenly fail. Yes you can mitigate that by pinning the version, but I doubt you want to do that and miss out on patches

I'd just write your own tests, even if it means lifting and dropping the code